### ISSUE TITLE
Android - Custom Marker component only renders partially sometimes

### ISSUE THREAD

**Issue by erik-lissen** [react-native-maps/react-native-maps#4636]
### Summary

We're using custom markers to render circles on the map.
The issue only seems to happen on android.
We are changing the array of markers based on user input, and sometimes only a quarter of the circle gets rendered.

Here's a screenshot of the issue:
![Screenshot_20230308-194701 (1)](https://user-images.githubusercontent.com/111526841/225660638-b15dbd74-11f4-437a-aaf2-5ad38c7df3cd.png)

It's supposed to be a full circle:
![CleanShot 2023-03-16 at 15 10 37](https://user-images.githubusercontent.com/111526841/225661039-305dde78-a59f-4f81-9632-8a9bfb86805b.png)

I haven't been able to pin-point the exact scenario when it happens, it seems to be completely random.

### Reproducible sample code

```TSX
//circles are rendered like this:
//currentData changes when the user taps on a circle - this causes a fetch, returning new markers

const markers = useMemo(() => {
    return currentData?.map((marker, index) => {
      const smallestUserCount = Math.min(
        ...(currentData?.map(m => m.uniqueListeners ?? 1) ?? []),
      );
      const largestUserCount = Math.max(
        ...(currentData?.map(m => m.uniqueListeners ?? 1) ?? []),
      );

      const opacities = [0.4, 0.6, 0.8];

      //map marker.uniqueListeners to a percentage based on the smallest and largest user count, then map that percentage to a range of opacities
      const opacity = (userCount: number) => {
        const percentage =
          (userCount - smallestUserCount) /
          (largestUserCount - smallestUserCount);
        const opacityIndex = Math.floor(percentage * (opacities.length - 1));
        const opacityEntry = opacities[opacityIndex];
        if (opacityEntry === undefined) return opacities[2];
        return opacityEntry;
      };
      return (
        <Circle
          key={marker.shortName}
          marker={marker}
          opacity={opacity(marker.uniqueListeners ?? 1)}
        />
      );
    });
  }, [currentData]);

//the map renders them like:
return (
    <>
      <MapView
        provider="google"
        maxZoomLevel={ZOOM_LEVELS[level]}
        zoomEnabled={false}
        ref={mapRef}
        style={{
          flex: 1,
          justifyContent: 'flex-end',
          alignItems: 'center',
        }}>
        {markers}
      </MapView>
    </>
  );
```

### Steps to reproduce

It's not clear how to reproduce - it doesn't happen very frequently.

### Expected result

The markers should be displaying properly

### Actual result

The markers are sometimes rendered incorrectly (only the top-left quarter of the circle gets drawn)

### React Native Maps Version

1.3.2

### What platforms are you seeing the problem on?

Android

### React Native Version

0.69.4

### What version of Expo are you using?

Not using Expo

### Device(s)

Pixel 3

**Comment by iboxgithub** [comment]
same issue for us

**Comment by devsales** [comment]
This happened to me as well and I fixed it by setting `width` and `height` for the `Marker` element. So something like this:
```JavaScript
<Marker
   style={{ width: 100px, height: 100px }}
   {...otherProps}
>
   <MyCustomMarker />
</Marker>
```
Basically it seems that on Android the marker has some predefined size and everything outside is cut off.

**Comment by erik-lissen** [comment]
Thanks for the suggestion, for us this did not fix it.

However - for anyone else who might read this - after some testing, I figured out that this happens when I try to change the styling of an existing marker on the map.

In our use case, we are rendering different 'levels' of markers on the map, the user can click on a marker to go one level deeper. Sometimes, on the next level, the marker is in the same location as the previous level, so using the lat/long as keys for the marker, the same marker gets re-rendered with new styling, and this causes it to break.

My fix was to figure out a unique key for every scenario, meaning we're rendering new markers rather than editing the same one.

**Comment by madebydavid** [comment]
We have an application with a user journey which causes this issue reliably.

We experimented with a bunch of things:

- Changing `tracksViewChanges` after a delay
- Forcing `tracksViewChanges` to always be on
- Calling the `redraw` function on the marker ref in a `setTimeout`
- Using an SVG image instead of the native `<Image />` component in our marker

None of these worked.

What we instead found was that there is a hard coded value in the `createDrawable` function [here](https://github.com/react-native-maps/react-native-maps/blob/v1.3.2/android/src/main/java/com/airbnb/android/react/maps/AirMapMarker.java#L518) which is being used instead of the correct size for the rendered custom marker.

```java
  private Bitmap createDrawable() {
    int width = this.width <= 0 ? 100 : this.width;
    int height = this.height <= 0 ? 100 : this.height;
    this.buildDrawingCache();

    // Do not create the doublebuffer-bitmap each time. reuse it to save memory.
    Bitmap bitmap = mLastBitmapCreated;
 ```

We found that if we changed the 100 to a larger value then the issue went away.
